File: LoginPage.js
const nameInput = document.querySelector('input[name="name"]');
const emailInput = document.querySelector('input[name="email"]');
const contactInput = document.querySelector('input[name="contact"]');
const userBtn = document.querySelector('button[value="user"]');
const adminBtn = document.querySelector('button[value="admin"]');

window.addEventListener('load', function() {
    const loggedIn = localStorage.getItem('loggedIn');
    if (loggedIn === 'user') {
        window.location.replace('ParkingSlot.html');
    } else if (loggedIn === 'admin') {
        window.location.replace('AdminPage.html');
    }
});

function validEmail(email) {
    return email.endsWith('@gmail.com');
}

function validName(name) {
    return !/\d/.test(name);
}

function validContact(contact) {
    return contact.length === 11;
}

contactInput.addEventListener('input', function(e) {
    let value = e.target.value.replace(/\D/g, '');
    if (value.length > 11) {
        value = value.substring(0, 11);
    }
    e.target.value = value;
});

userBtn.addEventListener('click', async function(e) {
    e.preventDefault();
    const name = nameInput.value.trim();
    const email = emailInput.value.trim();
    const contact = contactInput.value.trim();
    if (!name || !email || !contact) {
        alert('Please fill up all the fields.');
        return;
    }
    if (!validName(name)) {
        alert('Name cannot contain numbers.');
        return;
    }
    if (!validEmail(email)) {
        alert('Incorrect email.');
        return;
    }
    if (!validContact(contact)) {
        alert('Contact number must be exactly 11 digits.');
        return;
    }

    try {
        const response = await fetch('http://localhost:5000/login', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ name, email, contact, role: 'user' })
        });
        const result = await response.json();
        if (result.success) {
            localStorage.setItem('loggedIn', 'user');
            localStorage.setItem('userEmail', email);
            window.location.replace('ParkingSlot.html');
        } else {
            alert(result.message);
        }
    } catch (error) {
        alert('Error connecting to server. Please try again.');
    }
});

adminBtn.addEventListener('click', async function(e) {
    e.preventDefault();
    const name = nameInput.value.trim();
    const email = emailInput.value.trim();
    const contact = contactInput.value.trim();
    if (!name || !email || !contact) {
        alert('Please fill up all the fields.');
        return;
    }
    if (!validName(name)) {
        alert('Name cannot contain numbers.');
        return;
    }
    if (!validContact(contact)) {
        alert('Contact number must be exactly 11 digits.');
        return;
    }
    if (email !== 'admin123@gmail.com') {
        alert('Sorry, you are not the admin.');
        return;
    }

    try {
        const response = await fetch('http://localhost:5000/login', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ name, email, contact, role: 'admin' })
        });
        const result = await response.json();
        if (result.success) {
            localStorage.setItem('loggedIn', 'admin');
            window.location.replace('AdminPage.html');
        } else {
            alert(result.message);
        }
    } catch (error) {
        alert('Error connecting to server. Please try again.');
    }
});

File: LoginPage.py
def validate_email(email):
    """
    Validates if the email ends with @gmail.com.
    Returns True if valid, False otherwise.
    """
    if not email.endswith('@gmail.com'):
        print("incorrect email")
        return False
    return True

def validate_contact(contact):
    """
    Validates if the contact number is only digits and at most 11 characters.
    Returns True if valid, False otherwise.
    """
    if not contact.isdigit() or len(contact) > 11:
        print("invalid contact number")
        return False
    return True

def login(name=None, email=None, contact=None, submit_type=None):
    """
    Simulates the login process by taking user inputs and validating them.
    If arguments are provided, use them; otherwise, prompt for input.
    """
    if name is None:
        name = input("Name: ")
    if email is None:
        email = input("Email: ")
    if contact is None:
        contact = input("Contact Number: ")
    if submit_type is None:
        submit_type = input("Submit as 'user' or 'admin': ").lower()
    else:
        submit_type = submit_type.lower()

    # Validate email first
    if not validate_email(email):
        return  # Don't proceed

    # Validate contact
    if not validate_contact(contact):
        return  # Don't proceed

    # Check submit type
    if submit_type == 'admin':
        if email != 'admin123@gmail.com':
            print("sorry you are not the admin")
            return  # Don't proceed
        else:
            print("Admin login successful!")
    else:
        print("User login successful!")

if __name__ == "__main__":
    login()
